### PR TITLE
Added key value memory store.

### PR DESCRIPTION
**Pull request by errhal**

### CODE REVIEW

File: src/main/scala/services/QueryService.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      }
+    } else {
+      "Invalid query."
+    }
+  }
+
+  def getKeyValue(message: String): String = {
+
+    val keyValue = keyStoreSelectPattern.r.findAllIn(message)
+    DatabaseManager.getKey(keyValue.group(2), keyValue.group(1))
```
Comment (rafalh) [inline src/main/scala/services/QueryService.scala]: i would use constants for groups or variables to improve readability here

File: src/main/scala/stores/KeyValueMemStore.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+  * Created by Kamil Radziszewski on 08.11.17.
+  */
+object KeyValueMemStore {
+
+  val keyValueDb = new TrieMap[String, TrieMap[String, AnyRef]]
+
+  def putValue(data: String, key: String, value: AnyRef): Boolean = {
+    var dataKeyValue = keyValueDb.getOrElseUpdate(data, new TrieMap[String, AnyRef]())
```
Comment (rafalh) [inline src/main/scala/stores/KeyValueMemStore.scala]: you always create new TrieMap here even if one already exists. Its non optimal

File: src/main/scala/managers/DatabaseManager.scala [hunk 3]
```diff
@@ -2,22 +2,56 @@ package managers
...
-  def saveData(datafile:String, data: Any): Unit = {
+  val keyValueDbFile = "keyValue.db"
 
+  def initDbFiles(): Unit = {
+    initializeKeyValueDbFile()
   }
 
-  def loadData(datafile:String): JsonNode = {
+  def initializeKeyValueDbFile(): Unit = {
+    val dbFile = new File(ConfigurationProvider.getDbLocation() + "/" + keyValueDbFile)
```
Comment (rafalh) [inline src/main/scala/managers/DatabaseManager.scala]: you should use File.separator for portability

File: src/main/scala/managers/DatabaseManager.scala [hunk 4]
```diff
@@ -2,22 +2,56 @@ package managers
...
+  }
+
+  def saveData(datafile: String, data: Any): Unit = {
+
+  }
+
+  def loadData(datafile: String): JsonNode = {
     val dbFile = new File(ConfigurationProvider.getDbLocation() + "/" + datafile + ".db")
     val objectMapper = new ObjectMapper()
     val jsonNode = objectMapper.readValue(dbFile, classOf[JsonNode])
```
[the remark this answers is not in the document]
Comment (errhal) [reply src/main/scala/managers/DatabaseManager.scala]: I think for this moment, it is the best to use stable, good evolving library (like Jackson).

**Comment by errhal** [comment]
Thanks for review !

**Pull request merged**